.github/workflows/ci.yml

```diff
@@ -74,6 +74,40 @@ jobs:
       run: ctest --output-on-failure -C $BUILD_TYPE
 
 
+  # Job testing several versions of hdf5
+  # ===================================================
+  Linux_HDF5_Versions:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        hdf5_version : [ hdf5-1_8_22, hdf5-1_10_9, hdf5-1_12_2, hdf5-1_13_2 ]
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+    - name: "Install libraries"
+      run: |
+        sudo apt-get -qq update
+        sudo apt-get -qq install ninja-build libsz2 zlib1g-dev
+
+    - name: Build HDF5
+      run: |
+        git clone https://github.com/HDFGroup/hdf5 --depth 1 --branch ${{ matrix.hdf5_version }}
+        mkdir -p hdf5/BUILD && cd hdf5/BUILD
+        cmake .. -GNinja -DCMAKE_INSTALL_PREFIX=$PWD/I -DHDF5_ENABLE_Z_LIB_SUPPORT=ON -DHDF5_ENABLE_SZIP_SUPPORT=ON -DHDF5_ENABLE_SZIP_ENCODING=ON -DUSE_LIBAEC=ON
+        ninja && ninja install
```

Comment:
Just an alternative (that we use in neuron) in case building HDF5 from source would become problematic/too slow: 1) we can keep platform fixed (e.g. ubuntu 22.04) 2) build hdf5 library with system compiler and store the binary artefacts somewhere 3) use those artefacts here in the CI

Downside is we will need to rebuild HDF5 artefacts but that we will need once in a while when we change base OS image (which we don't need to do often).

Reply:
Or, just using one of the binary distribution that they provide with centos7/centos8 platform:
<img width="1311" alt="image" src="https://user-images.githubusercontent.com/666852/201205904-5274f0d2-4b38-4421-80d8-b4bb5badf6eb.png">